src/aliceVision/mesh/Mesh.cpp

```diff
@@ -2258,284 +2265,151 @@ void Mesh::getLargestConnectedComponentTrisIds(StaticVector<int>& out) const
+    while (!nodes.empty())
+    {   
+        aiNode * node = nodes.back();
+        nodes.pop_back();
+
+        Eigen::Matrix4d T;
+        for (int i = 0; i < 4; i++)
         {
-            if(line.size() < 3 || line[0] == '#')
-            {
-                // nothing to do
-            }
-            else if((line[0] == 's') && (line[1] == ' '))
-            {
-                // nothing to do
-            }
-            else if((line[0] == 'v') && (line[1] == ' '))
-            {
-                Point3d pt;
-                if(useColors)
-                {
-                    float r, g, b;
-                    sscanf(line.c_str(), "v %lf %lf %lf %f %f %f", &pt.x, &pt.y, &pt.z, &r, &g, &b);
-                    // convert float color data to uchar
-                    _colors.emplace_back(
-                      static_cast<unsigned char>(r*255.0f),
-                      static_cast<unsigned char>(g*255.0f),
-                      static_cast<unsigned char>(b*255.0f)
-                    );
-                }
-                else
-                {
-                    sscanf(line.c_str(), "v %lf %lf %lf", &pt.x, &pt.y, &pt.z);
-                }
-                pts.push_back(pt);
-            }
-            else if((line[0] == 'v') && (line[1] == 'n') && (line[2] == ' '))
+            for (int j = 0; j < 4; j++)
             {
-                Point3d pt;
-                sscanf(line.c_str(), "vn %lf %lf %lf", &pt.x, &pt.y, &pt.z);
-                // printf("%f %f %f\n", pt.x, pt.y, pt.z);
-                normals.push_back(pt);
+                T(i, j) = node->mTransformation[i][j];
             }
-            else if((line[0] == 'v') && (line[1] == 't') && (line[2] == ' '))
+        }
+
+        for (int idMesh = 0; idMesh < node->mNumMeshes; idMesh++)
+        {
+            unsigned int meshId = node->mMeshes[idMesh];
+
+            aiMesh * mesh = scene->mMeshes[meshId];
+
+            if (!mesh->HasFaces())
             {
-                Point2d pt;
-                sscanf(line.c_str(), "vt %lf %lf", &pt.x, &pt.y);
-                uvCoords.push_back(pt);
+                //Why should we have a mesh without faces ?
+                continue;
             }
-            else if((line[0] == 'f') && (line[1] == ' '))
+
+            for (int idPoint = 0; idPoint < mesh->mNumVertices; idPoint++)
             {
-                int n1 = mvsUtils::findNSubstrsInString(line, "/");
-                int n2 = mvsUtils::findNSubstrsInString(line, "//");
-                Voxel vertex, uvCoord, vertexNormal;
-                Voxel vertex2, uvCoord2, vertexNormal2;
-                bool ok = false;
-                bool withNormal = false;
-                bool withUV = false;
-                bool withQuad = false;
-                if(n2 == 0)
+                map_indices[idPoint] = pts.size();
+
+                aiVector3D v = mesh->mVertices[idPoint];
+                pts.push_back(Point3d(v.x, v.y, v.z));
```

Comment:
```suggestion
                pts.emplace_back(v.x, v.y, v.z);
```
requires to add the wrapper method to `StaticVector`

Reply:
I think adding this to `StaticVector` should work
```c++
    template< class... Args >
    void emplace_back( Args&&... args )
    {
        _data.emplace_back(std::forward<Args>(args)...);
    }
```